src/main/java/HW22DB/LessonDao.java

```diff
@@ -0,0 +1,123 @@
+package HW22DB;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class LessonDao {
+    private final Connection connection;
+
+    public LessonDao(Connection connection) {
+        this.connection = connection;
+    }
+
+    public void addLesson(Lesson lesson)  {
+        // INSERT INTO lesson (name1, updateAT )
+        // VALUES ("mathPro", "2023-01-23" );
+        try {
+            PreparedStatement statement = connection.prepareStatement("INSERT INTO lesson (name1, updateAT)\n" +
+                    "VALUES (?, ?)", Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, lesson.getName1());
+            statement.setDate(2, (Date) lesson.getUpdatedAt());
+
+            int rows = statement.executeUpdate();
+
+            if (rows == 0) {
+                throw new CustomerCreateException("Unable to create customer");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                lesson.setId(generatedKeys.getInt(1));
+            }
+
+        } catch (SQLException | CustomerCreateException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void delLesson(String name1) {
+        try {
+            PreparedStatement statement = connection.prepareStatement("DELETE FROM lesson WHERE name1=?");
+            statement.setString(1, name1);
+            statement.execute();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public List<Lesson> showAllLessons() {
+        Statement statement = null;
+        List<Lesson> lessons = new ArrayList<>();
+        try {
+            statement = connection.createStatement();
+            boolean dateReturned = statement.execute("SELECT * FROM lesson");
+            if (dateReturned) {
+                ResultSet resultSet = statement.getResultSet();
+                while (resultSet.next()) {
+                    Lesson lesson = new Lesson();
+                    lesson.setId(resultSet.getInt("id"));
+                    lesson.setName1(resultSet.getString("name1"));
+                    lesson.setUpdatedAt(resultSet.getDate("updateAT"));
+                    lesson.setHomework_id(resultSet.getInt("homework_id"));
+                    lessons.add(lesson);
+                }
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        return lessons;
+    }
+
+    public List<Object> showLessonByID(Integer id) {
```

Comment:
Returning type should be Lesson, not List<Object> (see my previous comment regarding entity relationship)